pkg/operator/sync_test.go

```diff
@@ -154,3 +160,103 @@ func withCABundle(caBundle string) kubeCloudConfigOption {
 		kubeCloudConfig.Data["ca-bundle.pem"] = caBundle
 	}
 }
+
+func TestReconcileSimpleContentAccessSecret(t *testing.T) {
+	masterPool := helpers.NewMachineConfigPool("master", nil, helpers.MasterSelector, "v0")
+	workerPool := helpers.NewMachineConfigPool("worker", nil, helpers.MasterSelector, "v0")
+	infraPool := helpers.NewMachineConfigPool("infra", nil, helpers.MasterSelector, "v0")
+	entitlementSecret := helpers.NewOpaqueSecret(ctrlcommon.SimpleContentAccessSecretName, ctrlcommon.OpenshiftConfigManagedNamespace, "abc")
+	workerEntitlementSecret := helpers.NewOpaqueSecret(ctrlcommon.SimpleContentAccessSecretName+"-"+workerPool.Name, ctrlcommon.MCONamespace, "abc")
+	infraEntitlementSecret := helpers.NewOpaqueSecret(ctrlcommon.SimpleContentAccessSecretName+"-"+infraPool.Name, ctrlcommon.MCONamespace, "abc")
+	outOfDateInfraEntitlementSecret := helpers.NewOpaqueSecret(ctrlcommon.SimpleContentAccessSecretName+"-"+infraPool.Name, ctrlcommon.MCONamespace, "123")
+
+	cases := []struct {
+		name               string
+		mcoSecrets         []*corev1.Secret
+		ocManagedSecrets   []*corev1.Secret
+		expectedMCOSecrets []corev1.Secret
+		layeredMCPs        []*mcfgv1.MachineConfigPool
+	}{
+		{
+			name:               "no entitlement secret on cluster, with opted-in pool",
+			ocManagedSecrets:   []*corev1.Secret{},
+			mcoSecrets:         []*corev1.Secret{},
+			expectedMCOSecrets: []corev1.Secret{},
+			layeredMCPs:        []*mcfgv1.MachineConfigPool{infraPool},
```

Comment:
So unfortunately, deep copying the pool caused some failures on the set functions since the two infra pool copies don't match exactly when we call `allPoolSet.Difference(layeredPoolSet)`. I added the `DeepCopy` to the remaining objects and parallelized the test function otherwise. Thanks for the tip with the shuffle command, that passed locally for me with these changes :smile:  

Perhaps, because we're not modifying the pools in these test cases, it's safe to use them in a parallel manner?

Reply:
> the two infra pool copies don't match exactly

Ahh, I know why: You passed the MCP list directly into `sets.New()` instead of making a set based upon the name of each MCP. So when you call `.DeepCopy()` on it, it would be a different object, affecting the equality comparison. That would explain the failure with adding `.DeepCopy()` to each of the MCP instances in the test. For example, try running this:

```golang
package main

import (
        "fmt"

        corev1 "k8s.io/api/core/v1"
        metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

func printPod(pod1, pod2 *corev1.Pod) {
        fmt.Printf("%s == %s ? %v\n", pod1, pod2, pod1 == pod2)
}

func main() {
        pod1 := &corev1.Pod{
                ObjectMeta: metav1.ObjectMeta{
                        Name: "pod-1",
                },
        }

        pod2 := &corev1.Pod{
                ObjectMeta: metav1.ObjectMeta{
                        Name: "pod-2",
                },
        }

        printPod(pod1, pod2)
        printPod(pod1, pod1.DeepCopy())
        printPod(pod1, pod1)
}
```

Reply:
My suggestion here would be to only consider the names of the MachineConfigPools to determine equality and then that would allow you to add `.DeepCopy()` to the test suite.

Reply:
Ahh that makes sense to me! I've updated with the name approach :smile: